CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/panes/HeadingPaneNGTest.java

```diff
@@ -131,16 +130,73 @@ public void testValidityChanged_disabled() {
     @Test
     public void testHierarchicalUpdate_paneIsNotQueryEnabled() {
         System.out.println("testHierarchicalUpdate_paneIsNotQueryEnabled");
+        final TestListener listener = new TestListener();
+
+        final HeadingPane listenedHeadingPane = new HeadingPane(headingText, pluginsList, listener, globalParamLabels);
+
+        assertFalse(listener.hasHierarchicalUpdated());
+        listenedHeadingPane.hierarchicalUpdate();
+        assertTrue(listener.hasHierarchicalUpdated());
+    }
+
+    /**
+     * Test heading pane sends message to listeners when the pane is enabled and
+     * the validity has changed.
+     */
+    @Test
+    public void testValidityChanged_paneIsQueryEnabled() {
```

Comment:
I think I'm a bit confused by this test and the next. I don't think that the function itself changes behaviour based whether the pane is query enabled or not (just the function that it calls that is being tested separately). The unit tests for this function that @Auriga2 had written I think are ok so I think this and the next aren't necessary

Reply:
No problem, I was under the impression you wanted the test to assert something. This solution enabled the use of asserts, which used with mocks for the boxes FlowPane would allow full testing of the method.

Reply:
Happy for you to close this

Reply:
I do ideally want tests to assert something if possible. Those 2 tests I mentioned do already have tests with an appropriate assert. There is still one other unit test you updated in this PR which I think I'm ok with merging in